src/go/k8s/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -222,6 +223,7 @@ func (r *RedpandaReconciler) Reconcile(c context.Context, req ctrl.Request) (ctr
 	}
 
 	rp, result, err := r.reconcile(ctx, rp)
+	r.reconcileDefluxed(ctx, rp)
```

Comment:
Do we want to move this call just after the finalizer addition so that we more or less ignore it completely (I believe the above `r.reconcile` call is what creates the `Helm*` resources)?

I think we'll also likely want it to return `result.Result{}, err`. So, maybe something like:

```go
if r.Defluxed(rp) {
  return r.reconcileDefluxed(ctx, rp)
}

// migration stuff here

rp, result, err := r.reconcile(ctx, rp)

// status patch stuff here

```

Reply:
I would love to do that, but the least amount of changes to inject suspend flag into HelmRelease and HelmRepository is to call reconcile. That's why I proposed the function just after previous reconcile. 

I can change that, so `reconcileDefluxed` would patch HelmRelease and HelmRepository WDYT?

Reply:
Just to circle back here, we talked about keeping the call to `reconcileDefluxed` after the `reconcile` call, and just adding some error checking on it.

Reply:
Oh, I already changed that :)